references/references.go

```diff
@@ -0,0 +1,150 @@
+/*
+Copyright 2023 Nephio.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package references
+
+
+// NamespaceReference is used to reference a particular krm intstance
+type NamespaceReference struct {
+	// Namespace is the namespace of the referent.
+	// This field is required when referring to a Namespace-scoped resource and
+	// MUST be unset when referring to a Cluster-scoped resource.
+	//
+	// +optional
+	Namespace *Namespace `json:"namespace,omitempty"`
+
+	// Name is unique within a namespace to reference a nfClass.
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Name string `json:"name"`
+}
+
+// PackageRevisionReference is used to reference a particular package revision.
+type PackageRevisionReference struct {
+	// Namespace is the namespace of the referent.
+	// This field is required when referring to a Namespace-scoped resource and
+	// MUST be unset when referring to a Cluster-scoped resource.
+	//
+	// +optional
+	Namespace *Namespace `json:"namespace,omitempty"`
+
+	// Repository is the name of the repository containing the package
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	RepositoryName string `json:"repository"`
+
+	// PackageName is the name of the package for the revision
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	PackageName string `json:"packageName"`
+
+	// Revision is the specific version number of the revision of the package
+	//
+	// +kubebuilder:validation:MinLength=1
+	// +kubebuilder:validation:MaxLength=253
+	Revision string `json:"revision"`
+}
+
+// GVKReference identifies an API object containing controller-specific
+// configuration resource within the cluster.
+type GVKReference struct {
```

Comment:
Are similar things available in K8s code?

Reply:
I will remove this. This is defined in Gateway API fair.

Reply:
Ok...though I am not *sure* we want to take a dependency on GW API. So if we need this we may want to keep it here. If multiple projects are using some similar stuff we may want to upstream it to somewhere in more core K8s.

Reply:
yes, but for now I believe I can remove it as we don't need it so far.